seatingchart.py

```diff
@@ -23,270 +23,221 @@ def working_dir_path(name, slug, extension):
-        if lseat and lseat not in assignments:
+def arrange_seat(slug: str, layout: str, title: str = None, output: str = "out", lefty: bool = False, debug: bool = False) -> None:
+    # 1) a list of seats, in order of preference, with optionally ignored blank lines and repeats allowed
+    #    basically tab/newline separated, with no difference between them
+    SEATS_IN_ORDER = os.path.join("layouts", layout + "_ordered.txt")
+    assert_file_exists(SEATS_IN_ORDER)
+
+    # 1.5) a list of lefty seats, in order of preference, with optioonally ignored blank lines and
+    #      repeats allowed, basically tab/newline separted with no difference between them
+    if lefty:
+        LSEATS_IN_ORDER = os.path.join(
+            "layouts", layout + "_lefty_ordered.txt")
+        assert_file_exists(LSEATS_IN_ORDER)
+
+    # 2) a CSV list of students, one per line.
+    # ie what you get if you download a gradebook from courseworks
+    STUDENT_LIST = working_dir_path("roster", slug, "csv")
+    assert_file_exists(STUDENT_LIST)
+
+    # 2.5) a CSV list of students, one per line.
+    # ie what you get if you download a gradebook from courseworks
+    if lefty:
+        LSTUDENT_LIST = working_dir_path("lefty_roster", slug, "csv")
+        assert_file_exists(LSTUDENT_LIST)
+
+    # two lists, of students to assign first and last
+    assign_first_path = working_dir_path("assign-first", slug, "txt")
+    assign_last_path = working_dir_path("assign-last",  slug, "txt")
+
+    ASSIGN_FIRST = assign_first_path if os.path.isfile(
+        assign_first_path) else "/dev/null"
+    ASSIGN_LAST = assign_last_path if os.path.isfile(
+        assign_last_path) else "/dev/null"
+
+    # photos_path = os.path.join("out", slug, slug + "_files")
+    photos_path = os.path.join(
+        "images")
+    csv_sum = os.path.join(output, "seat.csv")
+
+    # 3) a tsv file containing the format of the room
+    layout_path = os.path.join("layouts", layout + ".txt")
+    assert_file_exists(layout_path)
+    LAYOUT = layout_path
+
+    # outputs:
+    TITLE = title if title != None else f"{slug} Seating"
+
+    # a CSV student id ordered list of assigned seats
+    OUTPUT_CSV = working_dir_path("list", slug, "csv")
+
+    # a pretty HTML version of uni <-> seats
+    OUTPUT_HTML = working_dir_path("list", slug, "html")
+
+    # an html page with seat, student, and photo
+    OUTPUT_CHART = working_dir_path("chart", slug, "html")
+
+    # Now we're ready to assign seats
+    assignments = {}
+
+    with open(SEATS_IN_ORDER, "r") as f:
+        seats = f.readlines()
+    seats = [s for s in seats if s[0] != '#']  # Strip comments
+    seats = list(itertools.chain.from_iterable(
+        [z.strip().split() for z in seats]))
+
+    assign_last = [x.strip() for x in open(ASSIGN_LAST).readlines()]
+    assign_first = [x.strip() for x in open(ASSIGN_FIRST).readlines()]
+
+    students = [tuple(s) for s in csv.reader(
+        open(STUDENT_LIST))]  # Skip two header rows
```

Comment:
should be in a `with` block

Reply:
can also be on one line

Reply:
this also doesn't seem to be skipping header rows... that seems to have disappeared from the original version